pkg/engine/task/task_delete.go

```diff
@@ -30,7 +30,7 @@ func (dt DeleteTask) Run(ctx Context) (bool, error) {
 	}
 
 	// 3. - Delete them using the client -
-	err = delete(enhanced, ctx.Client)
+	err = deleteObj(enhanced, ctx.Client)
```

Comment:
Because `delete` shadows the builtin `delete` function to remove an entry from a map, used in task_apply.go:284

To keep it homogeneous I've renamed all those to `applyResource`, `patchResource` and `deleteResource`